pkg/clients/oss/oss.go

```diff
@@ -0,0 +1,204 @@
+/*
+
+ Copyright 2021 The Crossplane Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+package oss
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/aliyun/aliyun-oss-go-sdk/oss"
+	"k8s.io/klog/v2"
+
+	"github.com/crossplane/provider-alibaba/apis/oss/v1alpha1"
+)
+
+// ClientInterface will help fakeOSSClient in unit tests
+type ClientInterface interface {
+	Describe(name string) (*sdk.GetBucketInfoResult, error)
+	Create(bucket v1alpha1.BucketParameter) error
+	Update(name string, aclStr string) error
+	Delete(name string) error
+}
+
+// SDKClient is the SDK client for Bucket
+type SDKClient struct {
+	Client *sdk.Client
+}
+
+// NewClient will create Bucket client
+func NewClient(ctx context.Context, endpoint string, accessKeyID string, accessKeySecret string, stsToken string) (*SDKClient, error) {
+	var (
+		client *sdk.Client
+		err    error
+	)
+	if stsToken == "" {
+		client, err = sdk.New(endpoint, accessKeyID, accessKeySecret)
+	} else {
+		client, err = sdk.New(endpoint, accessKeyID, accessKeySecret, sdk.SecurityToken(stsToken))
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to crate Bucket client: %v", err)
+	}
+	klog.Info("successfully created Bucket client.")
```

Comment:
We typically try not to mix loggers, do we need these extra statements outside the logging interface of the controller?

Reply:
No, I removed all of them.